test/fluree/db/query/history_test.clj

```diff
@@ -959,3 +959,50 @@
                           (-> (get c "f:commit")
                               (update "f:data" select-keys ["f:t"])
                               (select-keys ["f:author" "f:txn" "f:data"]))))))))))
+
+(deftest ^:integration txn-annotation
+  (with-redefs [fluree.db.util.core/current-time-iso (fn [] "1970-01-01T00:12:00.00000Z")]
+    (let [conn         @(fluree/connect {:method :memory})
+          ledger-name  "annotationtest"
+          ledger       @(fluree/create conn ledger-name)
+          context      [test-utils/default-str-context "https://ns.flur.ee" {"ex" "http://example.org/ns/"}]
+
+          db0 (fluree/db ledger)
+          db1 (->> @(fluree/stage db0 {"@context" context
+                                       "insert" [{"@id" "ex:betty"
+                                                  "@type" "ex:Yeti"
+                                                  "schema:name" "Betty"
+                                                  "schema:age" 55}]})
+                   (fluree/commit! ledger)
+                   (deref))
+
+          db2 (->> @(fluree/stage db1 {"@context" context
+                                       "insert" [{"@id" "ex:freddy"
+                                                  "@type" "ex:Yeti"
+                                                  "schema:name" "Freddy"
+                                                  "schema:age" 1002}]}
+                                  {:annotation {"originator" "opts" "foo" "bar"}})
+                   (fluree/commit! ledger)
+                   (deref))
+
+          db3 (->> @(fluree/stage db2 {"@context" context
+                                       "insert"   [{"@id"         "ex:letty"
+                                                    "@type"       "ex:Yeti"
+                                                    "schema:name" "Leticia"
+                                                    "schema:age"  38}]
+                                       "opts" {"annotation" {"originator" "txn" "bar" "foo"}}})
+                   (fluree/commit! ledger)
+                   (deref))]
+      (is (= [{"f:txn" "fluree:memory://annotationtest/txn/5c54eee1fd6be197402fcd76b035cd6f42b55723f35c294992d76e6eea2cf874"}
+              {"f:txn" "fluree:memory://annotationtest/txn/6ab2e58c44af9db30feb461c212b3b566a35fdfe92d53b61018eab425ca0c342"
+               "f:annotation" {"originator" "opts" "foo" "bar"}}
+
+              {"f:txn" "fluree:memory://annotationtest/txn/e342e6a53d26b78b6222446eac29060c5237f54175e1d6639cb900103b729ccc"
+               "f:annotation" {"originator" "txn" "bar" "foo"}}]
+             (->> @(fluree/history ledger {:context        context
```

Comment:
could you break this query and formatting apart and bind them to descriptive names in the `let` binding vector?